controllers/nutanixmachine_controller.go

```diff
@@ -597,8 +563,14 @@ func (r *NutanixMachineReconciler) getOrCreateVM(rctx *nctx.MachineContext) (*nu
 		return nil, err
 	}
 
-	memorySize := rctx.NutanixMachine.Spec.MemorySize
-	memorySizeMib := GetMibValueOfQuantity(memorySize)
+	diskList, err := getDiskList(rctx)
+	if err != nil {
+		errorMsg := fmt.Errorf("failed to get the disk list to create the VM %s. %v", vmName, err)
+		rctx.SetFailureStatus(capierrors.CreateMachineError, errorMsg)
+		return nil, err
+	}
```

Comment:
are we planning to fix all the code codecov warnings? 

controllers/nutanixmachine_controller.go#L566-L570
Added lines #L566 - L570 were not covered by tests

Reply:
Adding code coverage here seems like it will require more refactoring and pulling in a new dependency of Prism Go Client. Given this change needs to make its way into a release-v1.3 branch, I'd like to refrain from making unnecessary changes that would cause problems with cherry-picking. I can offer to make a separate PR for the main branch which will ensure these lines get covered on the main branch.

Reply:
ok